pallets/amm/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Automatic Market Maker (AMM)
+//!
+//! Given any [X, Y] asset pair, "base" is the `X` asset while "quote" is the `Y` asset.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
+
+mod pool_structs;
+
+pub use pallet::*;
+use pool_structs::{AMMCurve, LiquidityProviderAmounts, Pool, StabilityPool};
+
+#[frame_support::pallet]
+mod pallet {
+    use crate::{AMMCurve, LiquidityProviderAmounts, Pool, StabilityPool};
+    use core::marker::PhantomData;
+    use frame_support::{
+        pallet_prelude::{StorageDoubleMap, StorageMap, StorageValue, ValueQuery},
+        traits::{EnsureOrigin, GenesisBuild, Get, Hooks, IsType},
+        Blake2_128Concat, PalletId, Parameter,
+    };
+    use orml_traits::MultiCurrencyExtended;
+    use parallel_primitives::{Amount, Balance, CurrencyId, Rate};
+    use sp_arithmetic::traits::BaseArithmetic;
+
+    #[pallet::call]
+    impl<T: Config<I>, I: 'static> Pallet<T, I> {}
+
+    #[pallet::config]
+    pub trait Config<I: 'static = ()>: frame_system::Config {
+        type Currency: MultiCurrencyExtended<
+            Self::AccountId,
+            CurrencyId = CurrencyId,
+            Balance = Balance,
+            Amount = Amount,
+        >;
+        type Curve: AMMCurve;
+        type Event: From<Event<Self, I>> + IsType<<Self as frame_system::Config>::Event>;
+        type LpFee: Get<Rate>;
+        type PalletId: Get<PalletId>;
+        type PoolId: Default + BaseArithmetic + Parameter;
+        type PoolManager: EnsureOrigin<Self::Origin>;
+        type StabilityPool: StabilityPool;
+        type TreasuryAccount: Get<Self::AccountId>;
+        type TreasuryFee: Get<Rate>;
+    }
+
+    #[pallet::error]
+    pub enum Error<T, I = ()> {
+    }
+
+    #[pallet::event]
+    pub enum Event<T: Config<I>, I: 'static = ()> {}
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config<I>, I: 'static = ()> {
+        pub exchange_rate: Rate,
+        pub phantom: PhantomData<(T, I)>,
+    }
+
+    #[cfg(feature = "std")]
+    impl<T: Config<I>, I: 'static> Default for GenesisConfig<T, I> {
+        fn default() -> Self {
+            GenesisConfig {
+                exchange_rate: Default::default(),
+                phantom: PhantomData,
+            }
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config<I>, I: 'static> GenesisBuild<T, I> for GenesisConfig<T, I> {
+        fn build(&self) {
+            ExchangeRate::<T, I>::put(self.exchange_rate);
+        }
+    }
+
+    #[pallet::hooks]
+    impl<T: Config<I>, I: 'static> Hooks<T::BlockNumber> for Pallet<T, I> {}
+
+    #[pallet::pallet]
+
+    pub struct Pallet<T, I = ()>(_);
+
+    /// The exchange rate from the underlying to the internal collateral
+    #[pallet::storage]
+    pub type ExchangeRate<T, I = ()> = StorageValue<_, Rate, ValueQuery>;
```

Comment:
As stated before, `ExchangeRate` is a parameter used by both `pallet-loans` and `pallet-amm`. So, to remove this duplicity it is possible to create a common pallet or move `ExchangeRate` to a runtime parameter.

Reply:
Could be a runtime parameter marked as a constant IMO